lib/kubeclient/common.rb

```diff
@@ -455,6 +444,78 @@ def self.restclient_read_timeout_option
 
     private
 
+    def do_rest_request(args)
+      uri = @api_endpoint.merge("#{@api_endpoint.path}/#{@api_version}/#{args[:path]}")
+      options = {
+        url: uri.to_s,
+        method: args[:method],
+        headers: (args[:headers] || {}).merge('params' => args[:params])
+      }.merge(rest_client_options)
+      options[:block_response] = args[:block] if args[:block]
```

Comment:
nitpick: To avoid confusion let's call it `block_response` everywhere.

Because `Request.execute` supports both regular ruby block and a `:block_response` param, with different semantics (regular block gets RestClient::Response after redirect/error processing/decoding).
https://github.com/rest-client/rest-client/blob/v2.0.2/lib/restclient/request.rb#L718-L726